.github/workflows/upload-artifacts.yaml

```
@@ -43,6 +43,27 @@ jobs:
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v4
 
+      - name: Prepare artifacts in staging-deploy directories
+        run: ./gradlew publish
+
+      - name: Upload artifacts to Maven Central Repository
+        env:
+          JRELEASER_NEXUS2_USERNAME: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
+          JRELEASER_NEXUS2_PASSWORD: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
```

These two should be the ones for `mavenCentral` (due to the Jreleaser specification), although we can use the same Maven Central token, I think.
https://jreleaser.org/guide/latest/reference/deploy/maven/maven-central.html#_environment

```suggestion
          JRELEASER_MAVENCENTRAL_USERNAME: ${{ secrets.MAVEN_CENTRAL_USERNAME }}
          JRELEASER_MAVENCENTRAL_PASSWORD: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
```